src/tests/purchase.test.ts

```diff
@@ -22,6 +36,89 @@ describe("purchase", () => {
     ]);
     expect(APIGetRequest).toHaveBeenCalledWith(expectedRequest);
   });
+
+  test("appends checkout session id to http urls", async () => {
+    vi.mocked(mount).mockImplementation((_component, options) => {
+      options.props?.onFinished({
+        redeemUrl: "https://example.com",
+      });
+      return vi.fn();
+    });
+
+    const purchases = Purchases.getSharedInstance();
+    const offerings = await purchases.getOfferings();
+    const packageToBuy = offerings.current?.availablePackages[0];
+
+    const result = await purchases.purchase({
+      rcPackage: packageToBuy!,
+    });
+
+    expect(result.redemptionInfo?.redeemUrl).not.toBeNull();
+
+    const redeemUrl = new URL(result.redemptionInfo!.redeemUrl!);
+
+    expect(redeemUrl.searchParams.get("trace_id")).toBe(
+      "c1365463-ce59-4b83-b61b-ef0d883e9047",
+    );
+    expect(redeemUrl.searchParams.get("checkout_session_id")).toBe(
+      "c1365463-ce59-4b83-b61b-ef0d883e9047",
+    );
+  });
+
+  test("appends checkout session id to https urls", async () => {
+    vi.mocked(mount).mockImplementation((_component, options) => {
+      options.props?.onFinished({
+        redeemUrl: "https://example.com",
+      });
+      return vi.fn();
+    });
+
+    const purchases = Purchases.getSharedInstance();
+    const offerings = await purchases.getOfferings();
+    const packageToBuy = offerings.current?.availablePackages[0];
+
+    const result = await purchases.purchase({
+      rcPackage: packageToBuy!,
+    });
+
+    expect(result.redemptionInfo?.redeemUrl).not.toBeNull();
+
+    const redeemUrl = new URL(result.redemptionInfo!.redeemUrl!);
+
+    expect(redeemUrl.searchParams.get("trace_id")).toBe(
+      "c1365463-ce59-4b83-b61b-ef0d883e9047",
+    );
+    expect(redeemUrl.searchParams.get("checkout_session_id")).toBe(
+      "c1365463-ce59-4b83-b61b-ef0d883e9047",
+    );
+  });
+
+  test("does NOT append params to other protocols", async () => {
```

Comment:
👍 I think this makes sense, no need to send it if it's going to be unused... If we decide we want to track it from the mobile SDKs we can think about adding it for these as well.